netkan/netkan/scheduler.py

```diff
@@ -48,6 +49,52 @@ def schedule_all_netkans(self):
         for batch in sqs_batch_entries(messages):
             self.client.send_message_batch(**self.sqs_batch_attrs(batch))
 
+    def cpu_credits(self, cloudwatch, instance_id, start, end):
+        stats = cloudwatch.get_metric_statistics(
+            Dimensions=[{'Name': 'InstanceId', 'Value': instance_id}],
+            MetricName='CPUCreditBalance',
+            Namespace='AWS/EC2',
+            StartTime=start.strftime("%Y-%m-%dT%H:%MZ"),
+            EndTime=end.strftime("%Y-%m-%dT%H:%MZ"),
+            Period=10,
+            Statistics=['Average'],
+        )
+        # A pass consumes around 40 credits, with an accrue rate of 24/hr.
+        # So running every 2 hours should see using just a touch less than
+        # we gain in that time period.
+        creds = 0
+        try:
+            creds = stats['Datapoints'][0]['Average']
+        except IndexError:
+            logging.error("Couldn't acquire CPU Credit Stats")
+        return int(creds)
+
+    def volume_credits(self, cloudwatch, instance_id, start, end):
+        client = boto3.client('ec2')
+        response = client.describe_volumes(
+            Filters=[{
+                'Name': 'attachment.instance-id',
+                'Values': [instance_id]
+            }]
+        )
+        # If we add a second gp2 volume, this may break
+        volume = list(filter(lambda x: x['VolumeType'] == 'gp2', response['Volumes']))[0]
+        volume_id = volume['Attachments'][0]['VolumeId']
+        stats = cloudwatch.get_metric_statistics(
+            Dimensions=[{'Name': 'VolumeId', 'Value': 'vol-02cdb3dfd4b2a69f9'}],
```

Comment:
For the other metric we retrieve the id, it looks like here we've hard coded it. Is there a way to retrieve it instead?

Reply:
Nope, I made a mistake when testing the code :grimacing: